analyzers/tests/SonarAnalyzer.Test/TestCases/GenericLoggerInjectionShouldMatchEnclosingType.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using Microsoft.Extensions.Logging;
+
+public class Correct : Base
+{
+    Correct(ILogger<Correct> logger) : base(logger) { }              // Compliant
+    Correct(ILogger<Correct> logger, ILogger<Correct> logger2) { }   // Compliant
+
+    Correct(Wrapper<Correct> logger) { }                             // Compliant
+    Correct(Wrapper<Wrong> logger) { }                               // Compliant
+    Correct(Wrapper<Correct> logger, Wrapper<Wrong> logger2) { }     // Compliant
+
+    Correct(ILogger<string> logger) { }                              // Noncompliant {{Update this logger to use its enclosing type.}}
+    //              ^^^^^^
+    Correct(ILogger<Wrong> logger) { }                               // Noncompliant
+    //              ^^^^^
+    Correct(ILogger<Base> logger) : base(logger) { }                 // Noncompliant, need exact type match
+    //              ^^^^
+    Correct(ILogger<Wrapper<Correct>> logger) { }                    // Noncompliant
+    //              ^^^^^^^^^^^^^^^^
+    Correct(ILogger<ILogger<Correct>> logger) { }                    // Noncompliant
+    //              ^^^^^^^^^^^^^^^^
+
+    Correct(ILogger<string> logger, ILogger<Wrong> logger2) { }
+    //              ^^^^^^ {{Update this logger to use its enclosing type.}}
+    //                                      ^^^^^ @-1 {{Update this logger to use its enclosing type.}}
+
+    Correct(ILogger<string> logger, ILogger<Correct> logger2, ILogger<ILogger<Correct>> logger3) { }
+    //              ^^^^^^ {{Update this logger to use its enclosing type.}}
+    //                                                                ^^^^^^^^^^^^^^^^ @-1 {{Update this logger to use its enclosing type.}}
+
+    Correct(Logger<Correct> logger, Logger<Wrong> logger2) { }
+    //                                     ^^^^^ {{Update this logger to use its enclosing type.}}
+
+    Correct(Logger logger) { } // Compliant FP , Logger is not a generic type
```

Comment:
Do you mean FN? I think this case is expected to be compliant. It's not a generic logger so the name is not wrong.

Reply:
fair point, thanks!